src/util/simplify_expr.cpp

```diff
@@ -1610,8 +1610,13 @@ exprt simplify_exprt::bits2expr(
   else if(type.id()==ID_array)
   {
     const array_typet &array_type=to_array_type(type);
+    const auto &size = array_type.size();
 
-    const std::size_t n_el = numeric_cast_v<std::size_t>(array_type.size());
+    if(!size.is_constant())
+      return nil_exprt();
+
+    const std::size_t n_el =
```

Comment:
:no_entry_sign: This name is an abbreviation and doesn't mean anything to me. I would call this `size`, the above `size_expr`